src/civis/workflows/_validate.py

```diff
@@ -0,0 +1,33 @@
+"""Validation for Civis Platform workflow definitions."""
+
+import jsonschema
+import yaml
+
+from ._schemas import WORKFLOW_SCHEMA
+
+
+class WorkflowValidationError(Exception):
+    pass
+
+
+def validate_workflow_yaml(wf_def: str, /) -> None:
+    """Validate a YAML-formatted workflow definition.
+
+    Parameters
+    ----------
+    wf_def : str
+        YAML-formatted workflow definition.
+
+    Raises
+    ------
+    WorkflowValidationError
+        If the workflow definition is invalid.
+    """
+    _validate_workflow(yaml.safe_load(wf_def))
```

Comment:
I'm intentionally making a distinction between the publicly exposed `validate_workflow_yaml` for a YAML file path versus the private `_validate_workflow` for a Python dict. If/when we add more workflow-related utilities under the new `civis.workflows` namespace, we'll likely need to validate a (dynamically created?) workflow represented as a Python dict, in which case `_validate_workflow` could be used. But for now, the main use case we definitely have is to apply validation to an already existing YAML file sitting on the disk.